encoding/thrift/outbound.go

```diff
@@ -122,6 +122,53 @@ func (c thriftClient) Call(ctx context.Context, reqMeta yarpc.CallReqMeta, reqBo
 	// 		return success, resMeta, err
 	// 	}
 
+	out := c.ch.GetUnaryOutbound()
+
+	treq, proto, err := c.buildTransportRequest(reqMeta, reqBody)
+	if err != nil {
+		return wire.Value{}, nil, err
+	}
+
+	tres, err := out.Call(ctx, treq)
+	if err != nil {
+		return wire.Value{}, nil, err
+	}
+
+	defer tres.Body.Close()
+	payload, err := ioutil.ReadAll(tres.Body)
+	if err != nil {
+		return wire.Value{}, nil, err
+	}
+
+	envelope, err := proto.DecodeEnveloped(bytes.NewReader(payload))
+	if err != nil {
+		return wire.Value{}, nil, encoding.ResponseBodyDecodeError(treq, err)
+	}
+
+	switch envelope.Type {
+	case wire.Reply:
+		return envelope.Value, meta.FromTransportResponse(tres), nil
+	case wire.Exception:
+		var exc internal.TApplicationException
+		if err := exc.FromWire(envelope.Value); err != nil {
+			return wire.Value{}, nil, encoding.ResponseBodyDecodeError(treq, err)
+		}
+		return wire.Value{}, nil, thriftException{
+			Service:   treq.Service,
+			Procedure: treq.Procedure,
+			Reason:    &exc,
+		}
+	default:
+		return wire.Value{}, nil, encoding.ResponseBodyDecodeError(
+			treq, errUnexpectedEnvelopeType(envelope.Type))
+	}
+}
+
+func (c thriftClient) buildTransportRequest(
+	reqMeta yarpc.CallReqMeta,
+	reqBody envelope.Enveloper) (
```

Comment:
Styling nit: When breaking across multiple lines like this, prefer to end with
a trailing comma and a newline.

```go
func (c thriftClient) buildTransportRequest(
    reqMeta yarpc.CallReqMeta,
    reqBody envelope.Enveloper,
) (*transport.Request, protocol.Protocol, error) {
    // ...
}
```